File: API/ruby/lib/ruby_blackjack/deck.rb
class Deck
	attr_reader :Shoe, :Discard

	def initialize(shoeSize = 1)
		@Shoe = Array.new
		@ShoeSize = shoeSize
		buildDeck() #<--  Without the parens, it just seems naked.
		@Shoe.shuffle!
	end

	def Deal
		return @Shoe.shift
	end

	private
		def buildDeck
			(1..@ShoeSize).each do |x|
				Card.Ranks.each do |r_val|
					Card.Suits.each do |s_val|
						val = r_val[:Value]
						altValue = r_val[:AltValue]
						rank = r_val[:Rank]
						@Shoe.push(Card.new(rank,s_val,val,altValue))
					end
				end	
			end
		end	
end

File: API/PHP/deck.php

<?php
	require_once 'card.php';
	class deck{
		public $Shoe;
		public $Discard;



		public function __construct($size = 1)
		{
			$this->Shoe = array();
			$this->buildDeck($size);
			shuffle($this->Shoe);	
		}

		public function Deal()
		{
			return array_shift($this->Shoe);
		}

		private function buildDeck($size)
		{
			for($i=0; $i<$size; $i++)
			{
				foreach(card::Ranks() as $r)
				{
					foreach(card::Suits() as $s)
					{
						array_push($this->Shoe,new Card($r,$s));
					}
				}
			}
		}
		
	}

	if(count(get_included_files()) <= 2){Test_Deck();}

	function Test_Deck()
	{
		$d = new deck();
		echo json_encode($d->Shoe);
	}
?>